### PR TITLE
Create login controller and implement Firebase user authentication

### PR DESCRIPTION
**Pull request by cdiazj19**
File Login:
- Login Controller, view and module. Also, login.services with firebase authentication.

Login view: 
screencapture-localhost-8100-1451485166826

### CODE REVIEW

File: npm-debug.log [hunk 1]
```diff
@@ -0,0 +1,134 @@
+0 info it worked if it ends with ok
```
Comment (rfrancoch) [inline npm-debug.log:1]: Add the logs to .gitignore

File: www/app/login/login.controller.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+          template: 'Not available in this prototype'  
+        }).then(function(res){
+          console.log('Show alert')
+        });
+      };
+
+      ctrlScope.login = function(){
+        console.log("login submit : " + ctrlScope.user);
+        AuthService.authWithPassword(ctrlScope.user)
+          .then(ctrlScope.authSuccess)
```
Comment (rfrancoch) [inline www/app/login/login.controller.js:22]: mmm, authSuccess method has a non optional parameter, I recommend use explicit call for more readable code

File: www/app/app.module.js [hunk 3]
```diff
@@ -1,13 +1,15 @@
...
+      'angular-jwt',
       'ui.router',
-      'budgetrentacar.home',
-      'budgetrentacar.form'
+   
+      'budgetrentacar.login',
+      'budgetrentacar.home'
     ]);
-
 })();
```
Comment (aleks279) [inline www/app/app.module.js:15]: Try not to repeat this in the future, git itself is telling us that there needs to be a newline at the end of EVERY FILE.

File: www/scripts/ionic/css/ionic.css [hunk 4]
```diff
@@ -8519,7 +8524,7 @@ progress {
   vertical-align: top;
   text-align: center;
   text-overflow: ellipsis;
-  font-size: 16px;
+  font-size: 2em;
   line-height: 42px;
   cursor: pointer; }
   .button:hover {
```
Comment (aleks279) [inline www/scripts/ionic/css/ionic.css:8530]: Leave a newline in-between classes, makes the code look prettier and easier to understand.
Also, don't close the brackets in the same line as the last statement, close is in a line of its own, e.g. :
- Wrong way (technically not wrong, but ugly)

```
.class { css-statement: value; }
```
- Right way

```
.css-class {
        css-rule: value;
}
```

File: www/index.html [hunk 5]
```diff
@@ -22,20 +15,31 @@
...
+    <script src="https://cdnjs.cloudflare.com/ajax/libs/ngStorage/0.3.6/ngStorage.min.js"></script>
+    <script src="https://cdn.rawgit.com/auth0/angular-jwt/master/dist/angular-jwt.js"></script>
+
+  </head>
+  <body ng-app="budgetrentacar">
+
+    <ion-nav-view></ion-nav-view>
+
     <!-- your app's js -->
     <script src="app/app.module.js"></script>
```
Comment (rfrancoch) [inline www/index.html:30]: I think is better put the imports in the head tag

File: www/styles/login.css [hunk 6]
```diff
@@ -0,0 +1,26 @@
+.logo-budget{
```
Comment (rfrancoch) [inline www/styles/login.css:1]: I recommend to use a single file for all required custom styles, also not changes in ionic.css

File: www/app/login/login.controller.js [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+        $ionicPopup.alert({
+          tittle: 'Budget Rent a Car',
+          template: 'Not available in this prototype'  
+        }).then(function(res){
+          console.log('Show alert')
+        });
+      };
+
+      ctrlScope.login = function(){
+        console.log("login submit : " + ctrlScope.user);
```
Comment (aleks279) [inline www/app/login/login.controller.js:20]: `console.log`s are cool when you're developing and working on your local box, you know, to find problems and stuff... but get rid of them before pushing to git.

NO CONSOLE LOGS ARE ALLOWED IN PRODUCTION.

**Pull request closed without merging**